woocommerce/woocommerce-ios
# [Mobile Payments] Add modal view to notify users of progress in payment collection

Closes #3976 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader ⚠️

We continue building our prototype (for reference, p91TBi-4q4-p2), this time, by adding some UI elements to provide feedback about the payment collection process.

The UI in this PR does not match the designs (p91TBi-4Af-p2). The actual designs will be implemented in #3976 . In the meantime, this provides some user facing feedback, and allow us to finalize the underlying layers, so that we can continue iterating. 


https://user-images.githubusercontent.com/2722505/115773522-6cb5e980-a37e-11eb-9771-d76d8b25c29a.mov

This PR begins where #4010 ends. 

## Changes
* Add a layer of indirection, an orchestrator, if you will, between OrderDetailsViewController and the actual UI we present modally. This orchestrator, called OrderDetailsPaymentAlerts will allow us to switch the implementation of the actual modal views to a completely different layout whenever we attack #3980 
* Add an extension on FancyAlertViewController to render the modal views. This does not provide the layout we want, and will be removed later on.
* Add icons and images to the project, to be used in the modal views.
* Removes some of the temporary alerts we had put in place in previous iterations of the prototype.

## How to test
Testing is a bit complicated because:

1. Requires setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. Requires an external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* At that point, tapping the "Print receipt" button will trigger a printout. The "Email receipt button" does nothing yet, and the "Do nothing button will dismiss the modal view

cc @Garance91540 , fyi. This is far from the designs, but this is the first step towards having real UI, that matches the designs. We will be improving the UI in subsequent iterations.

Update release notes:
